### PR TITLE
Merge 1.6 into 1.7

### PR DESCRIPTION
**Pull request by NicolasT**
Upmerge of changes in 1.6 to 1.7 since the last merge.

CI is running, but given the very limited number of changes (none functionally) and compilation passing on TravisCI, I expect no issues.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ install:
   - eval `opam config -env`
   - opam remote add incubaid-devel -v -y -k git git://github.com/Incubaid/opam-repository-devel.git
   - opam update -v -y
-  - opam install -q -v -y ssl conf-libev lwt camlbz2 camltc
+  - opam install -q -v -y ssl conf-libev lwt camlbz2 camltc.0.7.0
```
Comment (domsj) [inline .travis.yml]: I don't think we want that for the 1.7 branch?
Actually I don't expect it to build on CI.

File: jenkins/base/008_install_dev_env.sh [hunk 2]
```diff
@@ -9,9 +9,10 @@ eval `opam config env`
 opam update -y
 opam pin camltc none || true
 opam install -y ssl
+opam remove -y ounit
 opam install conf-libev
 opam install -y camlbz2
 opam install -y snappy
 opam install -y "lwt.2.4.3"
-opam install -y "camltc.999"
+opam install -y "camltc.0.7.0"
```
Comment (domsj) [inline jenkins/base/008_install_dev_env.sh:17]: Same as above

File: src/node/catchup.ml [hunk 3]
```diff
@@ -99,7 +99,7 @@ let head_saved_epilogue hfn tlog_coll =
       | None -> Lwt.return ()
       | Some head_i ->
         begin
-          Logger.info_f_ "head_i = %s" (Sn.string_of head_i) >>= fun () ->
+	  Logger.info_f_ "head_i = %s" (Sn.string_of head_i) >>= fun () ->
```
Comment (domsj) [inline src/node/catchup.ml]: This inserts a tab instead of spaces

File: src/node/catchup.ml [hunk 4]
```diff
@@ -144,7 +144,7 @@ let catchup_tlog (type s) ~stop me other_configs ~cluster_id (current_i: Sn.t) m
   Lwt.catch
     (fun () ->
        _with_client_connection mr_addresses copy_tlog >>= fun () ->
-       Logger.info_f_ "catchup_tlog completed"
+      Logger.info_f_ "catchup_tlog completed"
```
Comment (domsj) [inline src/node/catchup.ml]: Making indentation worse

**Comment by NicolasT** [comment]
Addressed the whitespace-related review comments. Will fix TravisCI.

**Pull request closed without merging**